File: C++/A1/042.cpp
#include <iostream>
#include <stdio.h>
#include <cmath>
using namespace std;
int main(){
    int x=0,y=0,d;
    string n;
    cin >> n;
    int len = n.length();
    for(int i=0;i<len;i++){
        switch (n[i])
        {
        case 'N':
            y+=1;
            break;
        case 'S':
            y-=1;
            break;
        case 'E':
            x+=1;
            break;
        case 'W':
            x-=1;
            break;
        
        default:
            break;
        }
    }
    printf("%d %d %d",x,y,d=abs(x)+abs(y));

    
}

File: Python/A1/042.py
A = input()
x,y = 0,0
for move in A:
    if move == "N":
        y += 1
    elif move == "S":
        y -= 1
    elif move == "E":
        x += 1
    elif move == "W":
        x -= 1
print(f"{x} {y} {abs(x) + abs(y)}")